src/persist-client/src/read.rs

```diff
@@ -303,6 +423,29 @@ where
         // because Listen replays them at the original fidelity.
         (updates, frontier)
     }
+
+    /// Returns a `HollowBatch` enriched with this metadata appropriate for
+    /// listening-style behavior.
+    #[instrument(level = "debug", skip_all, fields(shard = %self.machine.shard_id()))]
+    pub async fn listen_hollow_batch(&mut self) -> ReaderEnrichedHollowBatch<T> {
+        trace!("Listen::listen_hollow_batch frontier={:?} ", self.frontier);
+        // Hack: Keep this method `&self` instead of `&mut self` by cloning the
+        // cached copy of the state, updating it, and throwing it away
+        // afterward.
+        let batch = self.machine.clone().next_listen_batch(&self.frontier).await;
+        let new_frontier = batch.desc.upper().clone();
+        let r = ReaderEnrichedHollowBatch {
+            shard_id: self.machine.shard_id(),
+            as_of: self.as_of.clone(),
+            // The HollowBatch should behave as if the frontier of the machine
```

Comment:
yeah, this is probably a hint to us that we should find a new name for `frontier` (it was never a great name for what it was doing)

Reply:
Nice--had the same thought here.

Reply:
I came up with the name `until` instead of `frontier` in the context that this is used (as the reader metadata). idk if that's true/useful/idiomatic, but made some sense to me. very open to suggestions.